packages/contractkit/src/kit.ts

```diff
@@ -49,24 +51,47 @@ export class ContractKit {
     return { ...this._defaultOptions }
   }
 
+  addAccount(privateKey: string) {
+    addLocalAccount(this.web3, privateKey)
+  }
+
   setGasCurrencyAddress(address: Address) {
     this._defaultOptions.gasCurrency = address
   }
 
-  sendTransaction(tx: Tx): TransactionResult {
+  async sendTransaction(tx: Tx): Promise<TransactionResult> {
+    const gasFeeRecipient = await this.web3.eth.getCoinbase()
```

Comment:
we should avoid doing all this work if the tx is not locally signed. since the node will on that case fill the gap. Neither should we fail if it's not a locally signed tx